packages/contracts/src/utils/asset_wrapper.ts

```diff
@@ -85,8 +85,15 @@ export class AssetWrapper {
                         userAddress,
                     );
                 } else if (tokenOwner === userAddress && desiredBalance.eq(0)) {
-                    // Burn token
-                    await erc721Wrapper.burnAsync(assetProxyData.tokenAddress, assetProxyData.tokenId, userAddress);
+                    // Transfer token to someone else
+                    const userAddresses = await (erc721Wrapper as any)._web3Wrapper.getAvailableAddressesAsync();
+                    const nonOwner = _.find(userAddresses, a => a !== userAddress);
+                    await erc721Wrapper.transferFromAsync(
```

Comment:
What's wrong with burning? Seems more clear to me.

Reply:
One cannot set an allowance for a non-existent token. One can however set an allowance to a token owned by someone else. In the combinatorial tests, we permute through all kinds of maker/taker allowance/balance scenarios and in some, the maker has 0 balance, but allowance is set. This change is necessary to support these permutations.